doc/import.rst

```diff
@@ -0,0 +1,81 @@
+Importing SpikeInterface
+========================
+
+SpikeInterface allows for the generation of powerful and reproducible spike sorting pipelines.
+Flexibility is built into the package starting from import to maximize the productivity of
+the developer and the scientist. Thus there are three ways that SpikeInterface and its components
+can be imported:
+
+
+Importing by Module
+-------------------
+
+Since each spike sorting pipeline involves a series of often repeated steps, many of the developers
+working on SpikeInterface recommend importing in a module by module fashion. This will allow you to
+keep track of your processing steps (preprocessing, postprocessing, quality metrics, etc.). This can
+be accomplished by:
+
+.. code-block:: python
+
+    import spikeinterface as si
+
+to import the :code:`core` module followed by:
+
+.. code-block:: python
+
+    import spikeinterface.extractors as se
+    import spikeinterface.preprocessing as spre
+    import spikeinterface.sorters as ss
+    import spikinterface.postprocessing as spost
+    import spikeinterface.qualitymetrics as sqm
+    import spikeinterface.exporters as sexp
+    import spikeinterface.comparsion as scmp
+    import spikeinterface.curation as scur
+    import spikeinterface.sortingcomponents as sc
+    import spikeinterface.widgets as sw
+
+to import any of the other modules you wish to use.
+
+The benefit of this approach is that it is lighter and faster than importing the whole package and allows
+you to choose which of the modules you actually want to use. If you don't plan to export the results out of
+SpikeInterface then you don't have to :code:`import spikeinterface.exporters`. Additionally the documentation
+of the package is set-up in a modular fashion, so if you have a problem with the module :code:`spikeinterface.curation`,
+you will know to go to the :code:`curation` section of this documention. The disadvantage of this approach is
+that you have more aliases to keep track of.
+
+
+Flat Import
```

Comment:
Did anyone favor this option in the issue? I kind of forgot it. Could we have this to the end?  I am not sure this is what numpy or pandas do btw. Is that the case? They also have `np.linalg` in numpy at least.

Reply:
That's fair. It was perhaps an oversimplification to explain a point.

Reply:
@samuelgarcia , you like this sometimes right?

Reply:
Yes, I mean, I think that your mind is on the right place. For user what matters is that: "all the functions that I could use or want will be available without nesting" so `si.function1`, `si.function2`, `si.function3` will work I think your example is fine from that perspective as it conveys the right idea with a familiar example.

Mine was more a technical caveat by saying that I don't think numpy or pandas are putting ALL their stuff the way that we are doing it:

https://github.com/SpikeInterface/spikeinterface/blob/329197618a9b48ef876d1e8b8e79f07f4abf5e49/src/spikeinterface/full.py#L26

Reply:
Very true! They keep more functions more private. I can change the wording slightly so more technically users are put off, but keep the simplicity of the example. Thanks :)

Reply:
I do not understand the question. Do I like this flat import ? yes I use it almost always.

For me the comparison to numpy/pndas is not the best I think.
I would more compare to scipy many modules that **could be** imported flat.

Reply:
@samuelgarcia 
The question was if that somebody was using it. You have answered it already by saying you do. Thanks.